sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalogSuite.scala

```diff
@@ -450,6 +451,26 @@ abstract class SessionCatalogSuite extends PlanTest {
     }
   }
 
+  test("alter table add columns") {
+    withBasicCatalog { sessionCatalog =>
+      sessionCatalog.createTable(newTable("t1", "default"), ignoreIfExists = false)
+      val oldTab = sessionCatalog.externalCatalog.getTable("default", "t1")
+      sessionCatalog.alterTableSchema(TableIdentifier("t1", Some("default")),
+        oldTab.schema.add("c3", IntegerType))
+      val newTab = sessionCatalog.externalCatalog.getTable("default", "t1")
+      if (sessionCatalog.externalCatalog.isInstanceOf[InMemoryCatalog]) {
+        assert(newTab.schema.toString == oldTab.schema.add("c3", IntegerType).toString)
```

Comment:
`StructField` returned from `CatalogTable` has metadata field that contains some hive related information, which the manually added schema field does not have. So directly comparing 2 `StructType` returns false.  `toString` method of `StructField` does not include metadata field..  But I think this may not be a good way to test anyway.. I will modify this test case..